c/tskit/tables.c

```diff
@@ -10790,12 +10761,36 @@ tsk_table_collection_dumpf(
         goto out;
     }
 
-    /* All of these functions will set the kas_error internally, so we don't have
-     * to modify the return value. */
-    ret = tsk_table_collection_write_format_data(self, &store, options);
+    /* Write format data */
+    ret = tsk_generate_uuid(uuid, 0);
     if (ret != 0) {
         goto out;
     }
+
+    ret = write_table_cols(&store,
+        (write_table_col_t[]){
```

Comment:
Can we assign this to a local variable please? Took me quite a bit to figure out what this was doing.